components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -1009,7 +1009,7 @@ public static boolean jsFunction_addAPI(Context cx, Scriptable thisObj,
         String transport = getTransports(apiData);
 
         String tier = (String) apiData.get("tier", apiData);
-        if(tier == null || "".equals(tier.trim())){
+        if(StringUtils.isEmpty(tier.trim())){
```

Comment:
This can result in a NPE as tier.trim() is evaluated before checking for null. Please fix.